cmd/downloader.cpp

```diff
@@ -178,11 +198,25 @@ int main(int argc, char* argv[]) {
         db::RWAccess db_access(db);
 
         // Stages 1 & 2 - Headers and bodies downloading - example code
-        HeadersStage header_stage{shared_status, block_exchange};
-        BodiesStage body_stage{shared_status, block_exchange};
+        HeadersStage header_stage{shared_status, block_exchange, &node_settings};
+        BodiesStage body_stage{shared_status, block_exchange, &node_settings};
+
+        // Trap os signals
+        SignalHandler::init([&](int) {
+            log::Info() << "Requesting threads termination\n";
+            header_stage.stop();
+            body_stage.stop();
+            block_exchange.stop();
+            sentry.stop();
+        });
```

Comment:
Why a custom handler for `SignalHandler::init()` ?
It's risky: if in another place we set another handler this one is forgotten.
Better to change in `Stoppable` the method `is_stopping` and check there `SignalHandler::signalled()`

Reply:
I have a mixed feeling about this. I think SignalHandler::init() must be called only one time. Anyway I made a commit to follow your suggestion.

Reply:
Yes should be called only once at silkworm start. And this happens way before pointers to stages are available. 
Downloader binary is actually way more simple than that